### PR TITLE
Fix a typo of the context name of asm options

### PR DESCRIPTION
**Pull request by MakeNowJust**
I think this is a failure of copy-and-pasting.

### CODE REVIEW

File: src/compiler/crystal/syntax/parser.cr [hunk 1]
```diff
@@ -4709,7 +4709,7 @@ module Crystal
       intel = false
       while true
         location = @token.location
-        option = parse_string_without_interpolation("asm clobber")
+        option = parse_string_without_interpolation("asm option")
```
Comment (RX14) [inline src/compiler/crystal/syntax/parser.cr]: How about "options" instead of option?

Comment (MakeNowJust) [inline src/compiler/crystal/syntax/parser.cr]: Which sounds better error message: `interpolation not allowed in asm option` or `interpolation not allowed in asm options`? I'm not native speaker, so I don't know.

Comment (RX14) [inline src/compiler/crystal/syntax/parser.cr]: I'd say the latter, there are multiple options allowed so it's plural right?

Comment (MakeNowJust) [inline src/compiler/crystal/syntax/parser.cr]:
```crystal
asm("nop" : "in1"(foo1), "in2"(foo2)
          : "out1"(bar1), "out2"(bar2)
          : "clobber1", "clobber2"
          : "volatile", "intel")
#   each of  ^------ and ^---  is an asm option string.
```

However `"volatile, intel"` is invalid asm option string. Asm option string can be appeared multiple times with separator, but `parse_string_without_interpolation` parses only a string literal containing an asm option. I think keeping singular is better.

Comment (RX14) [inline src/compiler/crystal/syntax/parser.cr]: Yep, I agree in this case that singular sounds right here.

**Comment by mverzilli** [comment]
Thank you!

**Pull request merged**